### PR TITLE
DPAD-1393: Stop auto-advancement after Xth play

### PR DESCRIPTION
**Pull request by gundams-and-dragons**
https://fandom.atlassian.net/browse/DPAD-1393

### CODE REVIEW

File: src/experimental/players/DesktopPauseAfterPlayPlayer/DesktopPauseAfterPlayPlayer.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { DesktopArticleVideoPlayerProps } from 'jwplayer/types';
+import { DesktopArticleVideoPlayerContent } from 'jwplayer/players/DesktopArticleVideoPlayer/DesktopArticleVideoPlayer';
+import usePauseAfterPlays from 'experimental/utils/usePauseAfterPlays';
+import PlayerWrapper from 'jwplayer/players/shared/PlayerWrapper';
+
+const DesktopPauseAfterPlayPlayerContent: React.FC<DesktopArticleVideoPlayerProps> = ({ videoDetails }) => {
+	// TODO: set actual pause after play amount
+	usePauseAfterPlays(1);
```
Comment (p-forfa) [inline src/experimental/players/DesktopPauseAfterPlayPlayer/DesktopPauseAfterPlayPlayer.tsx:9]: We'll need to change this here to the actual amount specified for the different variants.

File: src/loaders/DesktopArticleVideoLoader.tsx [hunk 2]
```diff
@@ -8,11 +8,11 @@ import { shouldLoadUcpPlayer } from 'loaders/utils/shouldLoadPlayer';
 
 export { getVideoPlayerVersion } from 'loaders/utils/GetVersion';
 
-const desktopReskinnedExperiment = defineExperiment({
-	name: 'desktop-reskinned-player',
-	buckets: ['p'],
-	startDate: Date.parse('2022-12-05T08:00:00'),
-	endDate: Date.parse('2022-12-14T11:59:00'),
+const desktopPauseAfterPlayExperiment = defineExperiment({
```
Comment (p-forfa) [inline src/loaders/DesktopArticleVideoLoader.tsx:11]: Based on the comments in the ticket, we might need to create different experiments for this? I forgot how we did variants in Pathfinder. I don't think or experiment library supports it, so we might need to create three different experiments in three different buckets, and pass in the pauseAfterXth video variable to the new [device]PauseAfterPlayPlayers. I'm not sure if there's a different way to do it:

```
This one will be an A/B test with multiple variants

stop auto-advancement after the 3rd video

after 5th

after 10th
```

File: src/experimental/players/DesktopPauseAfterPlayPlayer/DesktopPauseAfterPlayPlayer.tsx (hunk 1 above)
Comment (gundams-and-dragons) [reply src/experimental/players/DesktopPauseAfterPlayPlayer/DesktopPauseAfterPlayPlayer.tsx:9]: Yeah, I put a todo just above it when we figure out which to test first.

File: src/loaders/DesktopArticleVideoLoader.tsx (hunk 2 above)
Comment (gundams-and-dragons) [reply src/loaders/DesktopArticleVideoLoader.tsx:11]: We  need to do that. Once I know which to test and when I will update the experiment code.

File: src/loaders/DesktopArticleVideoLoader.tsx [hunk 3]
```diff
@@ -8,11 +8,25 @@ import { shouldLoadUcpPlayer } from 'loaders/utils/shouldLoadPlayer';
...
 
-const desktopReskinnedExperiment = defineExperiment({
-	name: 'desktop-reskinned-player',
-	buckets: ['p'],
-	startDate: Date.parse('2022-12-05T08:00:00'),
-	endDate: Date.parse('2022-12-14T11:59:00'),
+const desktopPauseAfterThreePlaysExperiment = defineExperiment({
+	name: 'desktop-pause-after-three-plays-player',
+	buckets: ['q'],
+	startDate: Date.parse('2023-2-13T08:00:00'),
```
Comment (p-forfa) [inline src/loaders/DesktopArticleVideoLoader.tsx:14]: I would change these dates to:

`
startDate: Date.parse('2023-02-22T08:00:00'),
 endDate: Date.parse('2023-03-01T11:59:00'),
`

File: src/loaders/MobileArticleVideoLoader.tsx [hunk 4]
```diff
@@ -15,6 +15,27 @@ const mobileReskinnedExperiment = defineExperiment({
 	endDate: Date.parse('2023-02-20T11:59:00'),
 });
 
+const mobilePauseAfterThreePlaysExperiment = defineExperiment({
+	name: 'mobile-pause-after-three-plays-player',
+	buckets: ['q'],
+	startDate: Date.parse('2023-2-13T08:00:00'),
```
Comment (p-forfa) [inline src/loaders/MobileArticleVideoLoader.tsx:21]: Same thing here with the dates.

File: src/experimental/utils/usePauseAfterPlays.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { useEffect, useContext } from 'react';
+import { PlayerContext } from 'jwplayer/players/shared/PlayerContext';
+import { PlaylistItemPlayerEventData } from 'jwplayer/types';
+import JWEvents from 'jwplayer/players/shared/JWEvents';
+
+export default function usePauseAfterPlays(pauseAfter: number) {
+	const { player } = useContext(PlayerContext);
+
+	useEffect(() => {
```
Comment (p-forfa) [inline src/experimental/utils/usePauseAfterPlays.ts:9]: 👍

**Review by p-forfa — APPROVED** [review]

**Review by p-forfa — APPROVED** [review]
The new pause changes LGTM. Looks like there are some merge conflicts in the Loader files. Otherwise, LGTM.

**Review by p-forfa — APPROVED** [review]
Date changes look good.

**Pull request merged**